GravatarApp/Networking/GravatarURLSession.swift

```diff
@@ -0,0 +1,40 @@
+import Foundation
+import Gravatar
+
+/// Intercepts the network requests to perform common operations.
+final class GravatarURLSession: URLSessionProtocol, Sendable {
```

Comment:
I don't feel strong about the name `GravatarURLSession` but couldn't come up with something better.